lib/net.js

```diff
@@ -125,6 +125,10 @@ function normalizeArgs(args) {
   const arg0 = args[0];
   var options = {};
   if (typeof arg0 === 'object' && arg0 !== null) {
```

Comment:
If this is doing the same thing as the `if (arguments.length === 1 && Array.isArray(arguments[0]))` block in `Socket.prototype.connect`, maybe remove the similar logic there?